:PROPERTIES:
:ID:       3f780680-b759-4db0-a62b-939cbed9b4fc
:END:
#+title: Assembly Problem (bioinformatics)
= the problem of assembling a genome without a reference genome.
* a.k.a. "de novo assembly", "shotgun assembly"
* some terms
** [[id:5e163670-71f1-4070-ae0d-d6e6483d6c08][Coverage (Bioinformatics)]]
* some "Laws of Assembly (bioinformatics)"
** The "First Law of Assembly"
 If a *prefix* of one read is *similar* to
    a *suffix* of another
 then they might come from the same place.
** The "Second Law of Assembly"
 = More coverage => more and longer overlaps.
** The "Third Law of Assembly"
 Repeats make assembly hard.
*** [[id:87f4f1e8-bf6b-4bd3-89ff-bb431bcc94fc][Aim For Longer Reads (Bioinformatics).]]
* [[id:a5891da4-6c20-471d-ac42-c666090d1cee][The Overlap Graph]]
* [[id:f18d3823-02dd-4842-b8d5-4903b4889ca6][Shortest Common Superstring (Ses) Problem]]
* [[id:2863ca13-3bdc-4432-89f8-0d91fb899216][The De Bruijn Graph]]
* Handling messy graphs.
This applies to both overlap graph and De Bruijn graph algorithms.
** Sequencing errors generate dead ends and small disconnected sets.
Discard them.
** Some edges are redundant.
So discard them.
*** An example.
In an graph whose nodes are kmers,
A -6-> B and B -6-> C imply A -5-> C.
** Polyploidy generates fork-merge pairs.
The graph looks like a path that forks into two,
and then those two sides rejoin.
The two forks correspond to the mother's and father's versions.
** Repeats generate ambiguity.
So just make a note that it's not clear how many times they are repeated.
